core/sawtooth/client.py

```diff
@@ -779,24 +779,34 @@ def get_transaction_status(self, txnid, timeout=30):
             request.get_method = lambda: 'HEAD'
             opener = urllib2.build_opener(self.proxy_handler)
             response = opener.open(request, timeout=timeout)
-            code = response.getcode()
-            response.close()
-            if code == 200:
-                return transaction.Status.committed
-            elif code == 302:
-                return transaction.Status.pending
-            else:
-                return transaction.Status.unknown
 
         except urllib2.HTTPError as err:
             LOGGER.error('peer operation on url %s failed with response: %d',
                          url, err.code)
+            if err.code == 302:
+                return transaction.Status.pending
+
+            raise MessageException('operation failed '
```

Comment:
This changes the behavior of errors from returning unknown to an exception.  From a quick review of clients currently checking for 'unknown', the only place that looks potentially problematic is simulator.py, which removes validators upon MessageException and has different behavior for 'unknown'.